### ISSUE TITLE
Fill "version" file with something that can parsed to put as meta-data into files

### ISSUE THREAD

**Issue by ekluzek** [ESMCI/manage_externals#66]
I created https://github.com/ESCOMP/ctsm/issues/215 which I think really belongs here. If when manage_externals is run it would create a file called something like "version_info" this file could be parsed and put as meta-data into NetCDF files to describe what "version" the checkout points to. We currently use subversion keywords to do this sort of thing. The main utility of this is when you've checked out a frozen tag. So just giving what that tag is would be good. If the tag is a branch, it's obviously not as useful, but this is used when it's a tag. A far secondary thing would be for it to add some notes if the checkout is dirty. But, that's secondary, because when I rely on subversion keywords it's knowing that the sandbox is NOT dirty.

We also might need a cime change to make this automatically picked up in cime.

**Comment by bandre-ucar** [comment]
It's not clear to me that this is actually a manage_externals issue or if it is a clm issue. The determining factor would be whether clm plans to take advantage of the unique information that manage_externals provides. 

The unique information provided by manage_externals is ***full*** set of externals and the clean/dirty status for each of those. If clm isn't going to use all that info and just wants a light weight reference for pointing back to a version of clm, then it should be handled by clm's infrastructure.

**Comment by ekluzek** [comment]
I agree I'm not sure who should be responsible for this. And I'd actually extend the responsibility to cime and maybe cesm as well (as possibilities). I did think that since both CLM and CESM (and later CAM and other stand-alone model configs) will be using manage_externals, that if manage_externals does this -- no one else would have to.

I also think that this would be utilized for writing out NetCDF files for cases (history and restart), but it would be used by CLM tools when creating input files to use.

I think this needs more discussion. I'm also looking into the cime side of this for cases. And there are some issues that need to be filed there. So for it to be able to be utilized for cases, those things will have to be fixed first.

**Comment by gold2718** [comment]
@ekluzek, Is there any existing procedure for documenting a run from a dirty sandbox?
With CESM 1, a single svn tag URL contains all the information about what was checked out.
With CESM 2, the answer depends on how manage_externals was run.
I think @ekluzek is asking for manage_externals to drop something which could be used to fill this gap. If I got that right, I second the request.

**Comment by ekluzek** [comment]
@gold2718 the rundb was the only way to document cases from dirty sandboxes. But, I don't know that it's used anymore? Or is it?

I'm not personally interested in documenting dirty sandboxes, but I am very interested in making sure that we document clean checkouts. I add the dirty sandbox case, because I think it is something that manage_externals could easily do if it already handles the clean case.

Technically in both CESM1 and CESM2 if you don't change externals (either through SVN_EXTERNAL_DIRECTORIES for CESM2 or cesm.xml for CESM2) a single bit of info contains your information (URL for svn and SHA for git). I'd like to at least have that info. available for cases or tools to add that to documentation. Right now that's not being done. 

But, yes CESM2 depends on manage_externals, and you might as well add what it did to your top level SHA as documentation on your run. If manage_externals is the way that we communicate the SHA to cases and tools -- you might as well include it's information as well. So yes we are asking for the same thing.
